keanu-project/src/main/java/io/improbable/keanu/vertices/utility/PrintVertex.java

```diff
@@ -0,0 +1,42 @@
+package io.improbable.keanu.vertices.utility;
+
+import com.google.common.base.Preconditions;
+import io.improbable.keanu.annotation.ExportVertexToPythonBindings;
+import io.improbable.keanu.tensor.dbl.DoubleTensor;
+import io.improbable.keanu.vertices.NonProbabilistic;
+import io.improbable.keanu.vertices.NonSaveableVertex;
+import io.improbable.keanu.vertices.dbl.DoubleVertex;
+import io.improbable.keanu.vertices.dbl.KeanuRandom;
+import java.io.PrintStream;
+
+public class PrintVertex extends DoubleVertex implements NonProbabilistic<DoubleTensor>, NonSaveableVertex {
+    private PrintStream printStream = System.out;
+    private final DoubleVertex parent;
+
+    @ExportVertexToPythonBindings
+    public PrintVertex(DoubleVertex parent, final PrintStream printStream) {
```

Comment:
I think the superclass `OutputStream` would be more appropriate than `PrintStream` here.

`PrintStream` is a `FilterOutputStream`. From its javadoc:

> These streams sit on top of an already existing output stream (the <i>underlying</i> output stream) which it uses as its basic sink of data, but possibly transforming the data along the way or providing additional functionality.

We don't need/want this, we just want to write some text to an `OutputStream`.

Reply:
ok fair enough